File: team.go
package main

type Team int

const (
	Unassigned Team = -1
	Red        Team = 0
	Blue       Team = 1
)

func (t Team) GetOppositeTeam() Team {
	switch t {
	case Red:
		return Blue
	case Blue:
		return Red
	default:
		return Unassigned
	}
}

File: frontend/src/utils/team.ts
import { Team } from '@/types/player';
import type { Composer } from 'vue-i18n';

export function teamToString(team: Team, i18n: Composer) {
  switch (team) {
    case Team.Red:
      return i18n.t('teams.red');
    case Team.Blue:
      return i18n.t('teams.blue');
    case Team.Unassigned:
      return i18n.t('teams.unassigned');
    default:
      return '';
  }
}
